lib/routes/actionsController.js

```diff
@@ -23,21 +23,61 @@
 'use strict';
 
 var domain = require('domain'),
+    util = require('util'),
     actions = require('../models/actions'),
     myutils = require('../myutils'),
     config = require('../../config'),
-    logger = require('logops');
+    constants = require('../constants'),
+    logger = require('logops'),
+    recentAxn = {}, corrAge = {};
 
 function PostAction(req, resp) {
     var firedEvent,
         errorAxn,
         nextDomain,
-        delay;
+        delay,
+        correlator = req.headers[constants.CORRELATOR_HEADER],
+        rulesWithCorr,
+        errMsg,
+        now = Date.now(),
+        MAX_AGE_CORR = 15 * 60 * 1000;
 
     firedEvent = req.body || {};
     firedEvent.subservice = req.subservice;
     firedEvent.service = req.service;
 
+    if (correlator) {
+        // try detect loop
+        corrAge[correlator] = now;
+        rulesWithCorr = recentAxn[correlator];
+        if (!rulesWithCorr) { //There is no array yet for this correlator
+            rulesWithCorr = recentAxn[correlator] = [];
+        }
+        if (util.isArray(rulesWithCorr)) {
+            for (var i = 0; i < rulesWithCorr.length; i++) {
+                if (rulesWithCorr[i] === firedEvent.ruleName) {
+                    errMsg = util.format('check infinite loop for %s (%s) - the rule has not been triggered',
```

Comment:
Is this the same exact message we had with the "1st try"? It would be great, in order to match with (potentially existing) documentation and alarm rules from OPs people.